spec/mailers/producer_mailer_spec.rb

```diff
@@ -128,6 +128,33 @@
     expect(mail.body.encoded).to include(p1.name)
   end
 
+  context 'when flag preferred_show_customer_names_to_suppliers is true' do
+    let(:s1) { create(:supplier_enterprise, preferred_show_customer_names_to_suppliers: true) }
+
+    it "adds customer names table" do
+      expect(s1.preferred_show_customer_names_to_suppliers).to eq(true)
```

Comment:
This assertion (just line 135) seems unnecessary - we're setting it to be true right before this example.